tests/system/src/test/java/test/robot/javafx/scene/TooltipTest.java

```diff
@@ -0,0 +1,295 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.scene;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Tooltip;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.StackPane;
+import javafx.scene.robot.Robot;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import javafx.stage.Window;
+import javafx.util.Duration;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import test.util.Util;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class TooltipTest {
+    static CountDownLatch startupLatch = new CountDownLatch(1);
+    static CountDownLatch tooltipShownLatch;
+    static Robot robot;
+    static Button button;
+    static Tooltip tooltip;
+
+    static volatile Stage stage;
+    static volatile Scene scene;
+
+    private static void assertTooltipShowDelay(long tooltipShowTime, long expectedTime) {
+        assertTooltipShowDelay(tooltipShowTime, expectedTime, 50);
+    }
+
+    private static void assertTooltipShowDelay(long tooltipShowTime, long expectedTime, long maximumDifference) {
+        // To avoid any small timing error we rather check if the value is between.
+        long maximumTime = expectedTime + maximumDifference;
+
+        assertTrue(tooltipShowTime >= expectedTime, tooltipShowTime + " >= " + expectedTime);
+        assertTrue(tooltipShowTime <= maximumTime, tooltipShowTime + " <= " + maximumTime);
+    }
+
+    @BeforeAll
+    static void initFX() {
+        Util.launch(startupLatch, TestApp.class);
+    }
+
+    @AfterAll
+    static void exit() {
+        Util.shutdown();
+    }
+
+    @Test
+    void testDefaultTooltip() {
+        assertFalse(tooltip.isShowing());
+
+        long tooltipShowTime = waitForTooltip();
+
+        assertTrue(tooltip.isShowing());
+        assertTooltipShowDelay(tooltipShowTime, 1000);
+    }
+
+    @Test
+    void testCssStylesheetTooltip() {
+        scene.getStylesheets().add(getClass().getResource("tooltip.css").toExternalForm());
+
+        assertFalse(tooltip.isShowing());
+
+        long tooltipShowTime = waitForTooltip();
+
+        assertTrue(tooltip.isShowing());
+
+        // Since the stylesheet needs to be processed we need a bigger timeout here.
+        long expectedTime = 30;
+        long maximumTime = expectedTime + 100;
+
+        assertTooltipShowDelay(tooltipShowTime, expectedTime, maximumTime);
```

Comment:
The third argument is a maximum difference, not a max time. So unless there is some reason to add the expected time twice, I would just use a larger constant delta.. You might even be able to get rid of the third arg depending on what we eventually go with for the default max delta.